pkg/exporter/sumologicexporter/sender.go

```diff
@@ -312,7 +313,14 @@ func (s *sender) handleReceiverResponse(resp *http.Response) error {
 			errMsgs = append(errMsgs, fmt.Sprintf("errors: %+v", rResponse.Errors))
 		}
 
-		return fmt.Errorf("failed sending data: %s", strings.Join(errMsgs, ", "))
+		err := fmt.Errorf("failed sending data: %s", strings.Join(errMsgs, ", "))
+
+		if resp.StatusCode == http.StatusBadRequest {
+			// Report the failure as permanent if the server thinks the request is malformed.
+			return consumererror.NewPermanent(err)
+		}
```

Comment:
Shouldn't be for all 4XX?

Reply:
Good question. Definitely not for all, like 401 or 429 ;) But for example 404 seems to make sense.
The OTLP spec explicitly only specified 400: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#bad-data